packages/grpc-js-core/src/call-credentials.ts

```diff
@@ -2,33 +2,33 @@ import {map, reduce} from 'lodash';
 
 import {Metadata} from './metadata';
 
-export type CallMetadataGenerator =
-    (options: {}, cb: (err: Error|null, metadata?: Metadata) => void) =>
+export type CallMetadataGenerator<T extends {} = {}> =
+    (options: T, cb: (err: Error|null, metadata?: Metadata) => void) =>
         void;
 
 /**
  * A class that represents a generic method of adding authentication-related
  * metadata on a per-request basis.
  */
-export interface CallCredentials {
+export interface CallCredentials<T extends {} = {}> {
```

Comment:
The generic parameters here and on the `ChannelCredentials` aren't really necessary. They don't even seem to be used for the `ChannelCredentials`, and the `options` parameter for `generateMetadata` and `CallMetadataGenerator` should just be `{ service_url: string }`.

Reply:
Is it guaranteed that the only option accepted by `CallCredentials` is `service_url`? I thought that since `createFromMetadataGenerator` is exposed to users that it can take arbitrary options.

Reply:
That option is provided by `CallCredentials` to user callbacks, so we decide what fields are there.